### ISSUE TITLE
Incorrect order of operations

### ISSUE THREAD

**Issue by luanpotter** [luanpotter/parent-pom#2]
With the new change that allowed for clean install's without gpg stuff, the order of operations gets wrong. When deploying a sub-project simply with

```
    mvn deploy -Pdeploy
```

It generates the javadoc and source **after** the signing, and does not sign those, which gets refused at the central.

**Comment by rtesini** [comment]
We could move everything to the profiles...The "build" outside would be empty and the "dev" profile would have everything except the GDG...And the "deploy" profile would have EVERYTHING. What do you think ?

**Comment by luanpotter** [comment]
That's perfect. In fact, I realized we only need to generate sources and javadoc in deploy profile anyway (even makes the build faster). So I added that there, and there was nothing left for the 'dev' profile. Of course we can create it later, in case it becomes relevant again. Now, the children's poms are even better; take a look at [http-façade](https://github.com/luanpotter/http-facade)'s, for instance; you just need to add this parent pom and your project is deployable to maven central. I also rolled out a 1.4.1 release there, and it worked flawlessly, so I'm closing this issue.

Thanks immensely for the help, it has been a really, really long time since I wanted to get this working ([reference](https://stackoverflow.com/questions/36824032/proper-execution-phase-for-maven-gpg-plugin)).

\o/
